src/overlays/actors/ovl_En_Vm/z_en_vm.c

```diff
@@ -27,35 +33,480 @@ const ActorInit En_Vm_InitVars = {
+    }
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        this->skelAnime.animCurrentFrame = this->skelAnime.initialFrame;
+    }
+}
+
+void EnVm_SetupStun(EnVm* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000068, -1.0f, SkelAnime_GetFrameCount(&D_06000068), 0.0f, 2, 0.0f);
+    this->unk_260 = 0;
+    this->timer = 180;
+    this->unk_25E = this->unk_260;
+    this->unk_21C = 2;
+    this->beamScale.z = 0.0f;
+    this->beamScale.y = 0.0f;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+    EnVm_SetupAction(this, EnVm_Stun);
+}
+
+void EnVm_Stun(EnVm* this, GlobalContext* globalCtx) {
+    if (this->timer == 0) {
+        if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+            this->unk_25E++;
+            if (this->unk_25E == 3) {
+                EnVm_SetupWait(this);
+            } else {
+                if (this->unk_25E == 1) {
+                    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000068, 1.0f, 0.0f,
+                                         SkelAnime_GetFrameCount(&D_06000068), 2, 0.0f);
+                } else {
+                    this->timer = 10;
+                    this->skelAnime.animCurrentFrame = 0.0f;
+                    this->skelAnime.animPlaybackSpeed = 2.0f;
+                }
+            }
+        }
+    } else {
+        Math_SmoothScaleMaxMinS(&this->beamRot, 0, 10, 0x5DC, 0);
+        this->timer--;
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    }
+}
+
+void EnVm_SetupDie(EnVm* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000068, -1.0f, SkelAnime_GetFrameCount(&D_06000068), 0.0f, 2, 0.0f);
+    this->timer = 33;
+    this->unk_25E = this->unk_260 = 0;
+    this->unk_21C = 3;
+    this->beamScale.z = 0.0f;
+    this->beamScale.y = 0.0f;
+    this->actor.shape.unk_08 = -5000.0f;
+    this->actor.posRot.pos.y += 5000.0f * this->actor.scale.y;
+    this->actor.velocity.y = 8.0f;
+    this->actor.gravity = -0.5f;
+    this->actor.speedXZ = Math_Rand_ZeroOne() + 1.0f;
+    this->actor.posRot.rot.y = Math_Rand_CenteredFloat(65535.0f);
+    EnVm_SetupAction(this, EnVm_Die);
+}
+
+void EnVm_Die(EnVm* this, GlobalContext* globalCtx) {
+    EnBom* EnBom;
+
+    this->beamRot.x += 0x5DC;
+    this->headRotY += 0x9C4;
+    Actor_MoveForward(&this->actor);
+
+    if (--this->timer == 0) {
+        EnBom = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, this->actor.posRot.pos.x,
+                            this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0x6FF, 0);
+
+        if (EnBom != NULL) {
+            EnBom->timer = 0;
+        }
+
+        Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.posRot, 0xA0);
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnVm_CheckHealth(EnVm* this, GlobalContext* globalCtx) {
+    EnBom* EnBom;
+
+    if (Actor_GetCollidedExplosive(globalCtx, &this->colliderCylinder) != NULL) {
+        osSyncPrintf("hp down %d\n", --this->actor.colChkInfo.health);
```

Comment:
can you decrement before the call then use the variable on its own for the arg? its pretty weird to do the decrement in a print imo, but if its needed then oh well